### ISSUE TITLE
Delivering fingerprints over HTTPS is useless.

### ISSUE THREAD

**Issue by rugk** [LineageOS/lineage_wiki#7]
Delivering fingerprints over http://wiki.lineageos.org/verifying-builds.html makes no sense, as without HTTPS any attacker can tamper with the listed fingerprints. **This has nothing to do with being paranoid, this has to do with basic security!**

Without HTTPS you have no basic security and an attacker could inject ads, phishing pages, other fingerprints or instructions how the attacked user compromises its whole system. Thus please use HTTPS. It is a simple step, but it does much.

BTW: Also use it on all other sites. With Let's Encrypt this is really easy.

**Comment by anoxi** [comment]
I guess it should say "Delivering fingerprints over HTTP is useless". It seems website and wiki use github pages so https is currently not possible. Are there plans to use cloudflare universal ssl in front of github or move to own infrastructure?

**Comment by kn00tcn** [comment]
but if you look at the github url or manually open the github, it's the correct one, so it's not as bad, in a way you end up proving that the http wiki is not tampered

similarly, if you go straight to the dloads or get the mirror link from another site, the key check can prove if either the key or dload is tampered (as long as you keep using different sources for both, or compare all sources together)

of course, not the best idea to let the user realize this, they'll just read the text & assume it's correct, assume the urls are correct, etc

**Comment by anoxi** [comment]
Yes, currently you can safely verify build authenticy here https://github.com/lineageos/lineage_wiki/blob/master/pages/meta/verifying_builds.md, but that's neither linked on download.lineage.org nor on the wiki.
I'd suggest to document file verification somewhere on https://download.lineage.org/ and just put a note on the wiki that files can be verified on the download page.

**Comment by kn00tcn** [comment]
download links to the wiki which links to the github, i'm pretty sure it did for at least a few days (although i cant remember if i saw it earlier than 4 days ago, which is when you made the report)

ya i guess the footer should have both links or a verifying info page on the downloads domain

**Comment by anoxi** [comment]
Another idea here to get the github pages content on https: http://serverfault.com/questions/583374/configure-nginx-as-reverse-proxy-with-upstream-ssl
Basically have little boxes that proxy securely to github.
